src/System.Management.Automation/engine/remoting/commands/InvokeCommandCommand.cs

```diff
@@ -1193,20 +1212,36 @@ protected override void EndProcessing()
             {
                 if (ParameterSetName.Equals(InvokeCommandCommand.InProcParameterSet))
                 {
-                    if (_steppablePipeline != null)
+                    Version currentStrictModeVersion = Context.EngineSessionState.CurrentScope.StrictModeVersion;
+                    try
                     {
-                        _steppablePipeline.End();
+                        if (_strictmodeversion != null)
+                        {
+                            Context.EngineSessionState.CurrentScope.StrictModeVersion = _strictmodeversion;
+                        }
+                        
+                        if (_steppablePipeline != null)
+                        {
+                            _steppablePipeline.End();
+                        }
+                        else
+                        {
+                            ScriptBlock.InvokeUsingCmdlet(
+                                contextCmdlet: this,
+                                useLocalScope: !NoNewScope,
+                                errorHandlingBehavior: ScriptBlock.ErrorHandlingBehavior.WriteToCurrentErrorPipe,
+                                dollarUnder: AutomationNull.Value,
+                                input: _input,
+                                scriptThis: AutomationNull.Value,
+                                args: ArgumentList);
+                        }
                     }
-                    else
+                    finally
                     {
-                        ScriptBlock.InvokeUsingCmdlet(
-                            contextCmdlet: this,
-                            useLocalScope: !NoNewScope,
-                            errorHandlingBehavior: ScriptBlock.ErrorHandlingBehavior.WriteToCurrentErrorPipe,
-                            dollarUnder: AutomationNull.Value,
-                            input: _input,
-                            scriptThis: AutomationNull.Value,
-                            args: ArgumentList);
+                        if (_strictmodeversion != null)
```

Comment:
Assuming you use StrictModeVersion property and not field, per earlier suggestion you would do the following here:
```suggestion
                        if (StrictModeVersion != null)
```